pallets/hyperdrive/src/lib.rs

```diff
@@ -0,0 +1,257 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+mod types;
+
+use frame_support::{dispatch::Weight, traits::Get};
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    use core::{fmt::Debug, str::FromStr};
+    use frame_support::{
+        pallet_prelude::*,
+        sp_runtime::traits::{
+            AtLeast32BitUnsigned, Bounded, CheckEqual, MaybeDisplay, SimpleBitOps,
+        },
+    };
+    use frame_system::pallet_prelude::*;
+    use sp_arithmetic::traits::{CheckedRem, Zero};
+    use sp_runtime::traits::Hash;
+    use types::*;
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    #[pallet::without_storage_info]
+    pub struct Pallet<T, I = ()>(PhantomData<(T, I)>);
+
+    #[pallet::config]
+    pub trait Config<I: 'static = ()>: frame_system::Config {
+        type RuntimeEvent: From<Event<Self, I>>
+            + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+        /// The output of the `Hashing` function used to derive hashes of target chain state.
+        type TargetChainHash: Parameter
+            + Member
+            + MaybeSerializeDeserialize
+            + Debug
+            + MaybeDisplay
+            + SimpleBitOps
+            + Ord
+            + Default
+            + Copy
+            + CheckEqual
+            + sp_std::hash::Hash
+            + AsRef<[u8]>
+            + AsMut<[u8]>
+            + MaxEncodedLen;
+        /// The block number type used by the target runtime.
+        type TargetChainBlockNumber: Parameter
+            + Member
+            + MaybeSerializeDeserialize
+            + Debug
+            + MaybeDisplay
+            + AtLeast32BitUnsigned
+            + Default
+            + Bounded
+            + Copy
+            + sp_std::hash::Hash
+            + FromStr
+            + MaxEncodedLen
+            + TypeInfo
+            + Zero
```

Comment:
I do not think we need all these constraints.

Reply:
I took them from frame_system::Config and only added extra what I need. We will see what we need.